src/deluge/gui/views/view.cpp

```diff
@@ -1435,6 +1453,83 @@ void View::displayAutomation() {
 	}
 }
 
+/// if you've toggled showing the VU meter, and the mod encoders are controllable (e.g. affect entire on)
+/// and the current mod button selected is the volume/pan button
+/// render VU meter on the grid
+bool View::potentiallyRenderVUMeter(uint32_t whichRows, RGB image[][kDisplayWidth + kSideBarWidth],
+                                    uint8_t occupancyMask[][kDisplayWidth + kSideBarWidth]) {
+	if (displayVUMeter && view.activeModControllableModelStack.modControllable
```

Comment:
can you cache and reuse the image? The UI updates faster than the envelope follower will actually react